### PR TITLE
Julia: Update memory management

### PR DESCRIPTION
**Pull request by cortner**
I've done a few small things
- created a branch `julia` where all julia development will first be merged to and then make PRs to `main` from there. 
- I've switched the memory management model to something that is better supported by the Julia community and hopefully future-safe. It turns out that also got me a big performance gain for small basis sizes.

### CODE REVIEW

**Review by Luthaf — COMMENTED** [review]
Looks good, I'll try to fix the Python failures on CI

File: julia/LICENSE.md~HEAD
[the remark this answers is not in the document]
Comment (cortner) [reply julia/LICENSE.md~HEAD:1]: I can get rid of it. I thought it was somebody else putting it there.

[the remark this answers is not in the document]
Comment (cortner) [reply julia/LICENSE.md~HEAD:1]: ... done

[the remark this answers is not in the document]
Comment (Luthaf) [reply julia/LICENSE.md~HEAD:1]: Thanks! CI should have been fixed with https://github.com/lab-cosmo/sphericart/pull/111, can you merge/rebase `main` with this branch?

**Comment by cortner** [comment]
done

**Review by Luthaf — APPROVED** [review]
All good!

**Pull request merged**